### PR TITLE
chore: update protocol a few and add links to each protocol

### PR DESCRIPTION
**Pull request by KazuCocoa**
Update protocol from https://github.com/WebKit/webkit/tree/main/Source/JavaScriptCore/inspector/protocol

This does not change anything for existing commands since no new command is called. I have just added links to each protocol and add `disable` command and a few which might be interested in.

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
Thanks @KazuCocoa for looking into it. I hope you'd be finally able to document this stuff better.
I would be happy to understand:
- how to inspect/sniff actual protocol commands and add/edit them in this module
- how the protocol works in general (data models, set of supported commands, etc)

File: lib/protocol/index.js [hunk 1]
```diff
@@ -2,103 +2,183 @@ const OBJECT_GROUP = 'console';
...
   'Debugger.setAsyncStackTraceDepth': ['depth'],
   'Debugger.setBreakpointsActive': ['active'],
   'Debugger.setPauseForInternalScripts': ['shouldPause'],
   'Debugger.setPauseOnAssertions': ['enabled'],
   'Debugger.setPauseOnExceptions': ['state'],
   //#endregion
 
+  // https://github.com/WebKit/WebKit/blob/main/Source/JavaScriptCore/inspector/protocol/DOM.json
   //#region DOM DOMAIN
-  'DOM.getDocument': [],
```
Comment (mykola-mokhnach) [inline lib/protocol/index.js:39]: was this command removed?

Comment (KazuCocoa) [reply lib/protocol/index.js:39]: no, that was my mistake for domdebugger..

**Comment by KazuCocoa** [comment]
I just read https://github.com/WebKit/WebKit/blob/main/Source/JavaScriptCore/inspector/protocol and pasted some of them (and tweaked the order following the WebKit's definitions. Haven't checked each behavior well yet, but basically i modified or send a new command in https://github.com/appium/appium-remote-debugger/blob/05b686e8be80659d4abbd6c499287b41699d629a/lib/rpc/rpc-client.js#L455-L500 (mostly a real device in my case) with enabling https://github.com/appium/appium-remote-debugger/blob/05b686e8be80659d4abbd6c499287b41699d629a/lib/rpc/rpc-client.js#L39 .

(Original motivation was to find clearing the local storage stuff, but no good one though)

Each behavior is not the same, but similar to https://chromedevtools.github.io/devtools-protocol/. 


https://github.com/appium/appium-remote-debugger/blob/05b686e8be80659d4abbd6c499287b41699d629a/development-notes.md also helps for this. As the document, I also try to estimate each behavior based on https://chromedevtools.github.io/devtools-protocol/. Basically Devtools protocol has much features, so WebKit one is kind of subset.

File: package.json [hunk 2]
```diff
@@ -47,7 +47,7 @@
     "teen_process": "^2.0.0"
   },
   "scripts": {
-    "build": "rm -rf build && babel --out-dir=build/lib lib && babel --out-dir=build index.js",
+    "build": "rm -rf build && babel --out-dir=build/lib lib && babel --out-dir=build index.js && babel --out-dir=build/bin bin",
```
Comment (KazuCocoa) [inline package.json:50]: then, `inspect-safari` for a simulator may work. Potentially we need to tweak the bin additionally as another PR.

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-xcuitest-driver/pull/1455 is one way to inspect communications, and try some commands via appium

**Review by mykola-mokhnach — APPROVED** [review]

**Pull request merged**